eva-accession-release-automation/analyze_mapping_weight_impact/snpmapinfo_metadata.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2020 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from dbsnp_mirror_metadata import *
+
+
+def get_build_version_from_file_name(file_name):
+    return os.path.basename(file_name).split("_")[0].lower()
+
+
+def lookup_GCA_assembly(species_name, snpmapinfo_table_name, asm, metadata_connection_handle):
+    query = "select assembly_accession from dbsnp_ensembl_species.EVA2015_snpmapinfo_asm_lookup " \
+            "where database_name = '{0}' and snpmapinfo_table_name = '{1}' and assembly = '{2}' " \
+            "and assembly_accession <> 'unresolved' and assembly_accession is not null"\
+        .format(species_name, snpmapinfo_table_name, asm)
+    results = get_all_results_for_query(metadata_connection_handle, query)
+    if len(results) == 0:
+        raise Exception("Could not look up GCA accession for {0} table for the species {1}"
+                        .format(snpmapinfo_table_name, species_name))
+    return results[0][0]
+
+
+# Overweight SNPs - SNPs with mapping weight > 3
+def get_distinct_asm_with_overweight_snps_in_snpmapinfo_table(snpmapinfo_table_name, species_info):
+    with get_db_conn_for_species(species_info) as species_db_connection_handle:
+        asm_columns = get_snpmapinfo_asm_columns(species_info, snpmapinfo_table_name)
+        if "asm_acc" in asm_columns:
+            distinct_asm_query = "select distinct asm_acc || '.' || asm_version as assembly " \
+                                 "from dbsnp_{0}.{1}"
+            type_of_asm = "assembly_accession"
+        else:
+            distinct_asm_query = "select distinct assembly from dbsnp_{0}.{1}"
+            type_of_asm = "assembly_name"
+        distinct_asm_query += " where weight > 3 and assembly is not null order by 1"
```

Comment:
Did we agree  a definition of overweight as strictly greater than 3? This mean a mapping to 10 or more places but it does not take into account the ones that maps to 2-10 places. It would be worth having the impact assessment run as well on `mapping weight > 1` at least so we know how it affects the numbers.

Reply:
I was using NCBI's definition for overweight SNPs. But I agree that this can be parametrized so that the script can be run for different values,